File: 1-Patron/python/GestorDeExperiencia.py
import Experiencia

class GestorDeExperiencia():
    def __init__(self, experiencia):
        self.__experiencia = experiencia

    def subirExperiencia(self, incremento):
        return self.__experiencia.incTotal(incremento)
        
    def generarExperiencia(self):
        return Experiencia()

File: 1-Patron/php/GestorDeExperiencia.php
<?php
class GestorDeExperiencia
{
  private $experiencia;

  function __construct($experiencia)
  {
    $this->experiencia = $experiencia;
  }

  function subirExperiencia($incremento)
  {
    $this->experiencia->incTotal($incremento);
  }

  function generarExperiencia()
  {
    return new Experiencia()
  }
}
?>
